petastorm/spark/spark_dataset_converter.py

```diff
@@ -216,6 +230,18 @@ def make_tf_dataset(
             workers_count = 4
         petastorm_reader_kwargs['workers_count'] = workers_count
 
+        if 'dataset_url_or_urls' in petastorm_reader_kwargs:
+            raise ValueError('User cannot set dataset_url_or_urls argument.')
+
+        if 'transform_spec' in petastorm_reader_kwargs or \
+                'infer_schema_from_first_row' in petastorm_reader_kwargs:
+            raise ValueError('User cannot set transform_spec and infer_schema_from_first_row '
```

Comment:
Shall we also allow users to use `transform_spec`&`infer_schema_from_first_row`?  Keeping `transform_spec` would make it consistent with the rest of the petastorm library.

Reply:
I think the param `preprocess_fn` should cover the functionality of `transform_spec`, and it is easier to use (can auto inferring result schema), so I forbid the two params.